### PR TITLE
Refactor: extract routes in folder

### PR DESCRIPTION
**Pull request by matjack1**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

This is a followup PR to #128.

I'm proposing an intermediate version between the existing solution and what we have in #128.

We are not introducing objects but we are separating functions in files, so we retain a clean router and smaller route files.

I haven't any strong preference between this or #128, I like both :) Maybe this is just a bit simpler?

### CODE REVIEW

**Comment by allella** [comment]
@Haxified @timmyichen @slikts @ThomasRoest @Nekoniri @Zeko369 

Thoughts on this PR vs #128 vs the existing routes?

**Comment by Nekoniri** [comment]
Seems like a decent compromise!

File: server/routers/routes/sampleRoute.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default function(req, res) {
```
Comment (timmyichen) [inline server/routers/routes/sampleRoute.ts]: this uses a regular function def while the other uses arrow functions - should we use the same for consistency?

**Comment by timmyichen** [comment]
also LGTM, minus a minor consistency thing.  does this pass TS linting as well?  (we should add that to CI)

**Comment by vaibhavsingh97** [comment]
@timmyichen Will send PR for the same

File: server/routers/routes/sampleRoute.ts (hunk 1 above)
Comment (vaibhavsingh97) [reply server/routers/routes/sampleRoute.ts]: nice catch  👍 , @matjack1 can you please make changes?

File: server/routers/exampleRouter.ts [hunk 2]
```diff
@@ -1,23 +1,13 @@
...
-    await new Promise(resolve => setTimeout(resolve, 3000));
-
-    const { query } = req;
-    res.json({ message: 'Delayed response from the server', query });
-  }),
-);
+router.get('/api/sample-route', sampleRoute);
+router.get('/api/sample-async-route', sampleAsyncRoute);
 
 router.get('/api/sample-errored-route', () => {
```
Comment (timmyichen) [inline server/routers/exampleRouter.ts]: also, any reason why this also doesn't have its own separately defined handler? 🙃

**Comment by allella** [comment]
@slikts are you good with this approach?

**Comment by vaibhavsingh97** [comment]
@slikts, please review

**Comment by slikts** [comment]
It's terse, doesn't use inheritance and doesn't add an unnecessary abstraction, so the previous objections don't apply.

**Comment by matjack1** [comment]
Thank you all. I've updated the PR taking onboard the suggestions and the linting passes on my machine, I'll add it to CI in a separate PR if that's OK.

**Comment by vaibhavsingh97** [comment]
@slikts and @timmyichen thanks for the review 👍

**Comment by nik-john** [comment]
@all-contributors please add @matjack1   for code

**Comment by nik-john** [comment]
@all-contributors please add @matjack1 for code

**Pull request merged**